### PR TITLE
add example plotting widget with play stop

### PR DESCRIPTION
**Pull request by chdoig**
Including an example showing widgets together with animations in the plotting interface (using a background thread for updating data)

Related with #2452

Demo:
startstop

### CODE REVIEW

File: examples/plotting/server/line_animate_widget.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+logger = logging.getLogger(__name__)
+
+N = 80
+
+x = np.linspace(0, 4*np.pi, N)
+y = np.sin(x)
+
+output_server("line_animate")
```
Comment (damianavila) [inline examples/plotting/server/line_animate_widget.py]: you should here `line_animate_widget` to pass the tests...

File: examples/plotting/server/line_animate_widget.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                time.sleep(0.05)
+    except:
+        logger.exception("An error occurred")
+        raise
+
+# spin up a background thread with animation
+Thread(target=background_thread, args=(ds,)).start()
+
+# endlessly poll to check widgets
+cursession().poll_document(curdoc(), 0.5)
```
Comment (bryevdv) [inline examples/plotting/server/line_animate_widget.py]: Interactive response seems laggy. Any reason not to lower the polling time?

**Comment by bryevdv** [comment]
Tested example, works as expected. One minor comment otherwise LGTM.

File: examples/plotting/server/line_animate_widget.py (hunk 2 above)
Comment (chdoig) [inline examples/plotting/server/line_animate_widget.py]: No reason, I was just using the default http://bokeh.pydata.org/en/latest/docs/reference/plot_objects.html#bokeh.session.Session.poll_document.

I've changed it to 0.05.

**Pull request merged**